components/camel-jpa/src/main/java/org/apache/camel/component/jpa/JpaHelper.java

```diff
@@ -54,34 +57,47 @@ public static EntityManager getTargetEntityManager(
 
         // then try reuse any entity manager which has been previously created and stored on the exchange
         if (em == null && exchange != null) {
-            em = exchange.getProperty(JpaConstants.ENTITY_MANAGER, EntityManager.class);
+            em = getEntityManagerMap(exchange).get(getKey(entityManagerFactory));
         }
 
         if (em == null && useSharedEntityManager) {
```

Comment:
since getEntityManagerMap() will now return empty map, should we add a condition for empty check as well ?